src/scripts/bench.py

```diff
@@ -380,6 +461,61 @@ def bench_key_agreement_algo(openssl, botan, algo):
     return KeyAgreementBenchmarkResult(algo, sorted(kszs_ossl.intersection(kszs_botan)), openssl_results, botan_results)
 
 
+def gen_html_report(botan_version, openssl_version, cmdline):
+    html = """<!DOCTYPE html>
+    <html>
+    <head>
+    <title>Botan Benchmark HTML Report</title>
+    </head>
+    <body>
+    <h1>Botan Benchmark HTML Report</h1>
+    <h2>Environment</h2>
+    <p><b>Date: </b> {date}</p>
+    <p><b>Operating System:</b> {os_info}</p>
+    <p><b>Botan version:</b> {botan_version}</p>
+    <details>
+        <summary><b>Build Info</b></summary>
+        <p><pre>{build_h}</pre></p>
+    </details>
+    <p><b>OpenSSL version:</b> {openssl_version}</p>
+    <p><b>Invocation:</b>
+    <pre>{cmdline}</pre>
+    </p>
+
+    <h2>Results</h2>"""
+
+    for img in sorted(os.listdir('bench_html/img')):
+        html += '<img src="img/%s">' % img
+
+    html += """
+    </body>
+    </html>
+    """
+
+    os_info = platform.system()
+    os_info += ' ' + platform.release()
+    os_info += ' ('
+    os_info += platform.machine()
+    os_info += ')'
+
+    include_path = re.search(r'-I(.*)', run_command(['botan', 'config', 'cflags']).strip()).group(1)
+    build_h = 'N/A'
+
+    try:
+        with open(os.path.join(include_path, 'botan/build.h'), 'r', encoding='utf-8') as f:
+            build_h = f.read()
```

Comment:
One thing I noticed while playing around with the benchmarks. This part is incompatible with a user-defined botan path (i.e., `--botan-cli`). Great work otherwise!

Reply:
Good point, you're right. We try our best and ask the CLI where to find the headers. But even a local binary `./botan` created without installing it will output `/usr/local/include/botan-3`, the designated install location. So there is nothing I can image we can do about it.